usersession/userd/privileged_desktop_launcher_internal_test.go

```diff
@@ -0,0 +1,486 @@
+Name[bs]=Otvori novi prozor pomoću privremenog profila
+Name[ca]=Obre una finestra nova amb un perfil temporal
+Name[ca@valencia]=Obri una finestra nova amb un perfil temporal
+Name[de]=Ein neues Fenster mit einem temporären Profil öffnen
+Name[en_AU]=Open a New Window with a temporary profile
+Name[eo]=Malfermi novan fenestron portempe
+Name[es]=Abrir una ventana nueva con perfil temporal
+Name[et]=Ava uus aken ajutise profiiliga
+Name[eu]=Ireki leiho berria behin-behineko profil batekin
+Name[fi]=Avaa uusi ikkuna käyttäen väliaikaista profiilia
+Name[fr]=Ouvrir une nouvelle fenêtre avec un profil temporaire
+Name[gl]=Abrir unha nova xanela con perfil temporal
+Name[he]=פתיחת חלון חדש עם פרופיל זמני
+Name[hy]=Բացել նոր պատուհան ժամանակավոր հատկագրով
+Name[ia]=Aperi un nove fenestra con un profilo provisori
+Name[it]=Apri una nuova finestra con un profilo temporaneo
+Name[ja]=一時プロファイルで新しいウィンドウを開く
+Name[ka]=ახალი ფანჯრის გახსნა დროებით პროფილში
+Name[kw]=Egery fenester noweth gen profil dres prys
+Name[ms]=Buka Tetingkap Baru dengan profil sementara
+Name[nb]=Åpne et nytt vindu med en midlertidig profil
+Name[nl]=Nieuw venster openen met een tijdelijk profiel
+Name[pt_BR]=Abrir uma nova janela com um perfil temporário
+Name[ro]=Deschide o fereastră nouă cu un profil temporar
+Name[ru]=Открыть новое окно с временным профилем
+Name[sl]=Odpri novo okno z začasnim profilom
+Name[sv]=Öppna ett nytt fönster med temporär profil
+Name[ug]=ۋاقىتلىق سەپلىمە ھۆججەت بىلەن يېڭى كۆزنەك ئاچ
+Name[vi]=Mở cửa sổ mới với hồ sơ tạm
+Name[zh_CN]=以临时配置文件打开新窗口
+Name[zh_TW]=以暫時性個人身分開啟新視窗
+Exec=env BAMF_DESKTOP_FILE_HINT=/var/lib/snapd/desktop/applications/chromium_chromium.desktop /snap/bin/chromium --temp-profile
+`
+
+func existsOnMockFileSystem(desktop_file string) bool {
+	return strutil.ListContains(mockFileSystem, desktop_file)
+}
+
+func (s *privilegedDesktopLauncherInternalSuite) TestDesktopFileIDToFilenameSucceedsWithValidId(c *C) {
+	restore := userd.MockRegularFileExists(existsOnMockFileSystem)
+	defer restore()
+
+	var desktopIdTests = []struct {
+		id     string
+		expect string
+	}{
+		{"mir-kiosk-scummvm_mir-kiosk-scummvm.desktop", "/var/lib/snapd/desktop/applications/mir-kiosk-scummvm_mir-kiosk-scummvm.desktop"},
+		{"foo-bar-baz.desktop", "/var/lib/snapd/desktop/applications/foo-bar/baz.desktop"},
+		{"baz-foo-bar.desktop", "/var/lib/snapd/desktop/applications/baz/foo-bar.desktop"},
+	}
+
+	for _, test := range desktopIdTests {
+		actual, err := userd.DesktopFileIDToFilename(test.id)
+		c.Assert(err, IsNil)
+		c.Assert(actual, Equals, test.expect)
+	}
+}
+
+func (s *privilegedDesktopLauncherInternalSuite) TestDesktopFileIDToFilenameFailsWithInvalidId(c *C) {
+	restore := userd.MockRegularFileExists(existsOnMockFileSystem)
+	defer restore()
+
+	var desktopIdTests = []string{
+		"mir-kiosk-scummvm-mir-kiosk-scummvm.desktop",
+		"bar-foo-baz.desktop",
+		"bar-baz-foo.desktop",
+		"foo-bar_foo-bar.desktop",
```

Comment:
I think we need more "fuzzing" invalid ID's here such as (just off the top of my head):

- [ ] "---------foo-bar-baz.desktop"
- [ ] "foo-bar-baz.desktop-foo-bar"
- [ ] "not-a-dot-desktop"
- [ ] "以临时配置文件打开新-non-ascii-here-too.desktop" 

I realize that some of these fail only because the associated files don't exist, but still good to make sure that nothing explodes when provided with non-ascii input for example